python-client/giskard/scanner/performance.py

```diff
@@ -0,0 +1,320 @@
+
+        return res
+
+    def _calculate_meta(self, model, dataset):
+        true_target = dataset.df.loc[:, dataset.target].values
+        pred = model.predict(dataset)
+
+        loss_values = [
+            metrics.log_loss([true_label], [probs], labels=model.meta.classification_labels)
+            for true_label, probs in zip(true_target, pred.raw)
+        ]
+
+        return pd.DataFrame({"__gsk__loss": loss_values}, index=dataset.df.index)
+
+    def _find_slices(self, dataset, model, meta: pd.DataFrame, slicer_name):
+        df_with_meta = dataset.df.join(meta)
+        target_col = "__gsk__loss"
+
+        # @TODO: Handle this properly once we have support for metadata in datasets
+        column_types = dataset.column_types.copy()
+        column_types["__gsk__loss"] = "numeric"
+        dataset_with_meta = Dataset(df_with_meta, target=dataset.target, column_types=column_types)
+
+        # Columns by type
+        cols_by_type = {
+            type_val: [col for col, col_type in dataset.column_types.items() if col_type == type_val]
+            for type_val in ["numeric", "category", "text"]
+        }
+
+        # Numerical features
+        slicer = self._get_slicer(slicer_name, dataset_with_meta, target_col)
+
+        slices = []
+        for col in cols_by_type["numeric"]:
+            slices.extend(slicer.find_slices([col]))
+
+        # Categorical features
+        slicer = CategorySlicer(dataset_with_meta, target=target_col)
+        for col in cols_by_type["category"]:
+            slices.extend(slicer.find_slices([col]))
+
+        # @TODO: FIX THIS
+        # Text features
+        slicer = TextSlicer(dataset_with_meta, target=target_col)
+        for col in cols_by_type["text"]:
+            slices.extend(slicer.find_slices([col]))
+
+        # @TODO: Should probably bind back to original dataset, but how to to pass
+        # cleanly also the metadata needed for the plots? Maybe need a wrapper object
+        # like Issue or similar.
+
+        return slices
+
+    def _get_slicer(self, slicer_name, dataset, target):
+        if slicer_name == "opt":
+            return OptSlicer(dataset, target=target)
+        if slicer_name == "tree":
+            return DecisionTreeSlicer(dataset, target=target)
+        if slicer_name == "ms":
+            return MultiscaleSlicer(dataset, target=target)
+        if slicer_name == "bf":
+            return BruteForceSlicer(dataset, target=target)
+
+        raise ValueError(f"Invalid slicer `{slicer_name}`.")
+
+
+class PerformanceScanResult(ScanResult):
+    def __init__(self, issues):
+        self.issues = issues
+
+    def has_issues(self):
+        return len(self.issues) > 0
+
+    def __repr__(self):
+        if not self.has_issues():
+            return "<PerformanceScanResult (no issues)>"
+
+        return f"<PerformanceScanResult ({len(self.issues)} issue{'s' if len(self.issues) > 1 else ''})>"
+
+    def _repr_html_(self):
+        tab_header = f"""
```

Comment:
as an improvement, moving it to a separate html template and using either a standard python string.Template or something like jinja could be a good idea

Reply:
This will definitely be rewritten with `jinja2` templates, but right now we are rushing. Also the `<iframe>` thing is a hack and will need to be changed. The best thing will be to wrap everything in a jupyter widget, but has been deprioritized for now.